src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
But what is it there we need to analyze? In `putcmdline()` there is a branch:

     if (!ui_is_external(kUICmdline)) {
         // do stuff
    } else {
        // do other stuff
    }

`ui_ext_cmdline_show` makes sense because the code is long. But otherwise why should `other stuff"` be in a separate function when `stuff` isn't?